core/cas-server-core-authentication-attributes/src/main/java/org/apereo/cas/services/AbstractRegisteredServiceAttributeReleasePolicy.java

```diff
@@ -97,6 +97,9 @@ public void postLoad() {
         val principalAttributes = resolveAttributesFromPrincipalAttributeRepository(principal, registeredService);
         LOGGER.debug("Found principal attributes [{}] for [{}]", principalAttributes, principal.getId());
 
+        LOGGER.trace("Finding requested attribute definitions");
+        getRequestedDefinitions().forEach(a -> principalAttributes.putIfAbsent(a, List.of()));
```

Comment:
I understand why this happens, but this will not work long-term. We should find a separate construct to do this; adding attributes with an immutable and empty list will cause problems.